src/client/pages/Dashboard.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState, FC, useEffect } from 'react'
+import { Container, Flex, Grid, VStack } from '@chakra-ui/react'
+
+import { ApiClient } from '../api'
+import { Navbar, CreateNew, CheckerCard } from '../components/dashboard'
+import { Checker } from '../../types/checker'
+
+export const Dashboard: FC = () => {
+  const [checkers, setCheckers] = useState<Checker[]>([])
+
+  const loadCheckers = async () => {
```

Comment:
Sorry for introducing so many libs...but [react-query](https://react-query.tanstack.com/quick-start) can help here. Provides out-the-box loading/error status. 

```ts
const { isLoading, isError, data: checker } = useQuery('checkers', () => ApiClient.get('/c))

// Render spinner whenever fetching...
{ isLoading && <Spinner />}
```

Then in other components that that needs to refetch checkers, you can use [query invalidation](https://react-query.tanstack.com/guides/query-invalidation):
```ts
const queryClient = useQueryClient()
queryClient.invalidate('checkers')
```